src/object/bonus_block.cpp

```diff
@@ -139,7 +138,7 @@ BonusBlock::BonusBlock(const ReaderMapping& mapping) :
   if (m_contents == Content::LIGHT || m_contents == Content::LIGHT_ON)
   {
     SoundManager::current()->preload("sounds/switch.ogg");
-    m_lightsprite = Surface::from_file("/images/objects/lightmap_light/bonusblock_light.png");
+    m_lightsprite = m_sprite->create_linked_sprite("on-light");
```

Comment:
This `m_lightsprite` name still hasn't been changed. In my opinion, if you are required to load & allocate another sprite, just to control when to draw it, then this system is heavily flawed. At least I think that's what's going on...

Reply:
What this is doing is only creating the "on-light" sprite when it's needed, which is only if the bonus block contains light.